### ISSUE TITLE
Connectivity plugin stream is not listening events.

### ISSUE THREAD

**Issue by sureace** [flutter/flutter#73920]
Hi,
     Connectivity plugin is not listening events. It is getting triggered only once when i launch the activity. But when i tried similar plugin like ConnectivityPlus, it was working. But it has build issues. Due to that i want to use connectivity plugin.

Pubspec.yaml
----------------
connectivity: ^2.0.2

connectivityStreamSubscription = Connectivity().onConnectivityChanged.listen((ConnectivityResult result) {
      print("Internet changed : $result");
    });

Thanks,
suresh

**Comment by briankariuki** [comment]
Please try this, reach out if you need more help

https://pub.dev/packages/connectivity

``` 
//Create connectivity class to access anywhere
class MyConnectivity {
  MyConnectivity._internal();

  static final MyConnectivity _instance = MyConnectivity._internal();

  static MyConnectivity get instance => _instance;

  Connectivity connectivity = Connectivity();

  StreamController controller = StreamController.broadcast();

  Stream get myStream => controller.stream;

  void initialise() async {
    ConnectivityResult result = await connectivity.checkConnectivity();
    _checkStatus(result);

    connectivity.onConnectivityChanged.listen((result) {
      _checkStatus(result);
    });
  }

  void _checkStatus(ConnectivityResult result) async {
    bool isOnline = false;

    if (result != ConnectivityResult.none) {
      try {
        final result = await InternetAddress.lookup('example.com');
        print(result);
        if (result.isNotEmpty && result[0].rawAddress.isNotEmpty) {
          isOnline = true;
        } else
          isOnline = false;
      } on SocketException catch (_) {
        isOnline = false;
      } on Error catch (_) {
        isOnline = false;
        print(_);
      }
    } else {
      isOnline = false;
    }
    controller.sink.add({result: isOnline});
  }

  void disposeStream() => controller.close();
 }

//In stateful widget import your connectivity class and do this
MyConnectivity _connectivity = MyConnectivity.instance;

//Call in initState
@override
  void initState() {
    super.initState();

    _connectivity.initialise();
    _connectivity.myStream.listen((source) {
      // setState(() {
      //   //do something
      // });
      String string;
      switch (source.keys.toList()[0]) {
        case ConnectivityResult.none:
          string = "Offline";
          break;
        case ConnectivityResult.mobile:
          string = "Mobile";
          break;
        case ConnectivityResult.wifi:
          string = "WiFi";
      }

      print(string);
      print(source.values.toList()[0]);
    });
  }


  //dispose stream
  @override
  void dispose() {
    _connectivity.disposeStream();
    super.dispose();
  }
```

**Comment by darshankawar** [comment]
@sureace 
Try above code and see if it works.
If issue still persists for you, provide `flutter doctor -v` and a complete minimal reproducible code sample along with expected and actual behavior.
Thanks.

**Comment by sureace** [comment]
Hi @darshankawar ,
              Thanks for your quick response and support. I tried as you said, but i am not getting change events in the realtime. I tried to put in cubit as well as put it directly in stateful widget. 
Source code
-------------
class _HomeMobilePortraitState extends State<HomeMobilePortrait> {
  MyConnectivity _connectivity = MyConnectivity.instance;

  @override
  void initState() {
    super.initState();
    _connectivity.initialise();
    _connectivity.myStream.listen((source) {
      // setState(() {
      //   //do something
      // });
      String string;
      switch (source.keys.toList()[0]) {
        case ConnectivityResult.none:
          string = "Offline";
          break;
        case ConnectivityResult.mobile:
          string = "Mobile";
          break;
        case ConnectivityResult.wifi:
          string = "WiFi";
      }

      print("Status Updated: $string");
      print(source.values.toList()[0]);
    });

  }

  @override
  void dispose() {
    _connectivity.disposeStream();
    super.dispose();
  }
}

class InternetCubit extends Cubit<InternetState> {
  MyConnectivity connectivity;

  InternetCubit({@required this.connectivity}) : super(InternetLoading()) {
    connectivity.initialise();
    connectivity.myStream.listen((source) {
      switch (source.keys.toList()[0]) {
        case ConnectivityResult.none:
          emitInternetDisconnected();
          break;
        case ConnectivityResult.mobile:
          emitInternetConnected(ConnectionType.Mobile);
          break;
        case ConnectivityResult.wifi:
          emitInternetConnected(ConnectionType.Wifi);
      }
    });
  }

  void emitInternetConnected(ConnectionType connectionType) =>
      emit(InternetConnected(connectionType: connectionType));

  void emitInternetDisconnected() => emit(InternetDisconnected());

  @override
  Future<void> close() {
    connectivity.disposeStream();
    return super.close();
  }
}

Flutter doctor
---------------
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E287 darwin-x64,
    locale en)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.52.1)
[✓] Connected device (1 available)

• No issues found!

Thanks

**Comment by darshankawar** [comment]
@sureace 
Cubit seems to be a 3rd party plugin. Please provide a complete minimal reproducible code sample without any 3rd party plugins that shows the issue you are facing.
Thanks.

**Comment by sureace** [comment]
@darshankawar 
As i said, i tried in both ways like using cubit and stateful widget. But i am not getting any change event when i toggle wifi, mobile status in a mobile. Same code is working fine with connectivity_plus plugin but i am getting build issues to use it.

class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return HomeMobileLandscape();
  }
}

class HomeMobilePortrait extends StatefulWidget {
  @override
  _HomeMobilePortraitState createState() => _HomeMobilePortraitState();
}

class _HomeMobilePortraitState extends State<HomeMobilePortrait> {
  MyConnectivity _connectivity = MyConnectivity.instance;

  @override
  void initState() {
    super.initState();
    _connectivity.initialise();
    _connectivity.myStream.listen((source) {
      // setState(() {
      //   //do something
      // });
      String string;
      switch (source.keys.toList()[0]) {
        case ConnectivityResult.none:
          string = "Offline";
          break;
        case ConnectivityResult.mobile:
          string = "Mobile";
          break;
        case ConnectivityResult.wifi:
          string = "WiFi";
      }

      print("Status Updated: $string");
      print(source.values.toList()[0]);
    });

  }

  @override
  void dispose() {
    _connectivity.disposeStream();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Home"),
      ),
      body: Center(
        child: Container(child:Text("Home screen"))
      ),
    );
  }
}


import 'dart:async';
import 'dart:io';

import 'package:connectivity/connectivity.dart';

class MyConnectivity{
  MyConnectivity._internal();

  static final MyConnectivity _instance = MyConnectivity._internal();

  static MyConnectivity get instance => _instance;

  Connectivity connectivity = Connectivity();
  StreamController controller;
  Stream get myStream => controller.stream;

  void initialise() async {
    controller = StreamController.broadcast();

    ConnectivityResult result = await connectivity.checkConnectivity();
    _checkStatus(result);

    connectivity.onConnectivityChanged.listen((result) {
      _checkStatus(result);
    });
  }

  void _checkStatus(ConnectivityResult result) async {
    bool isOnline = false;

    if (result != ConnectivityResult.none) {
      try {
        final result = await InternetAddress.lookup('google.com');
        print(result);
        if (result.isNotEmpty && result[0].rawAddress.isNotEmpty) {
          isOnline = true;
        } else
          isOnline = false;
      } on SocketException catch (_) {
        isOnline = false;
      } on Error catch (_) {
        isOnline = false;
        print(_);
      }
    } else {
      isOnline = false;
    }
    controller.sink.add({result: isOnline});
  }

  void disposeStream() => controller.close();

}

Thanks.

**Comment by darshankawar** [comment]
@sureace 
What's the expected behavior from your code sample ?
I see below when I run it:

```
I/flutter (23983): [InternetAddress('172.217.160.174', IPv4), InternetAddress('2404:6800:4009:80a::200e', IPv6)]
I/flutter (23983): Status Updated: WiFi
I/flutter (23983): true
I/flutter (23983): [InternetAddress('172.217.160.174', IPv4), InternetAddress('2404:6800:4009:80a::200e', IPv6)]
I/flutter (23983): Status Updated: WiFi
I/flutter (23983): true

```
Turning OFF wifi should print status as `false` ? and vice-versa ?

**Comment by sureace** [comment]
@darrenaustin ,
           I am not seeing any print statement like  "Status Updated: WiFi" or "Status Updated: Mobile" or  "Status Updated: Offline" 
Expected behavior
--------------------
Application need to track WiFi and Mobile Status, When any one is turned on, i need to show home screen. If both are turned off, then i have to show NoNetwork screen to the user. This has to happen in the realtime. Right now, i am getting the state only once (initially) and i need to reload the entire widget.

Thank you.

**Comment by darshankawar** [comment]
@sureace 
Can you try the official `connectivity` plugin [example](https://pub.dev/packages/connectivity/example) and check if you get expected behavior ?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
